fairlearn/reductions/_moments/conditional_selection_rate.py

```diff
@@ -49,15 +49,18 @@ def load_data(self, X, y, event=None, **kwargs):
         self.index = signed.index
         self.default_objective_lambda_vec = None
 
-        # fill in the information about the basis
         event_vals = self.tags[_EVENT].unique()
         group_vals = self.tags[_GROUP_ID].unique()
+        # The matrices pos_basis and neg_basis contain a lower-dimensional description of
+        # constraints, which is achieved by removing some redundant constraints.
+        # Including all constraints would make the vectors linearly dependent.
```

Comment:
Let's remove line 56--the relationship that we are going after is a bit more complicated than just linear dependence (that's why I'm just calling it `redundance` here). Also, we could have kept all the constraints to begin with--things would be correct--only less efficient. So if you want to give some additional clarification, maybe say:
```
# Considering fewer constraints is not required for correctness, but it can dramatically speed up GridSearch.
```

Reply:
Done! Thanks for elaborating